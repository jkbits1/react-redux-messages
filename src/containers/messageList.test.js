import React from 'react'
import { shallow } from 'enzyme'

import MessageList from './messageList'

it('renders correctly with no messages or members', () => {
  const props = {
    messages: [],
    membersLookup: {}
  }
  const tree = shallow(<MessageList {...props} />)
  expect(tree).toMatchSnapshot()
})

it('renders correctly with messages and no members', () => {
  const props = {
    messages: [
      {
        id: 'cd445e6d-e514-424f-ba8f-16ec842002c6',
        userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
        message:
          'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
        timestamp: '2017-02-09T04:27:38Z',
        readableTime: '4:02:00 AM',
        readableDate: '9th February 2017'
      },
      {
        id: 'b03569ae-ccbf-4975-8040-4daba638b407',
        userId: '16373df5-da0a-4074-8295-f916b94269f4',
        message:
          'Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.',
        timestamp: '2016-11-09T05:04:58Z',
        readableTime: '5:11:00 AM',
        readableDate: '9th November 2016'
      }
    ],
    membersLookup: {}
  }

  const tree = shallow(<MessageList {...props} />)
  expect(tree).toMatchSnapshot()
})

it('renders correctly with messages and members', () => {
  const props = {
    messages: [
      {
        id: 'cd445e6d-e514-424f-ba8f-16ec842002c6',
        userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
        message:
          'Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.',
        timestamp: '2017-02-09T04:27:38Z',
        readableTime: '4:02:00 AM',
        readableDate: '9th February 2017'
      },
      {
        id: 'b03569ae-ccbf-4975-8040-4daba638b407',
        userId: '16373df5-da0a-4074-8295-f916b94269f4',
        message:
          'Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.',
        timestamp: '2016-11-09T05:04:58Z',
        readableTime: '5:11:00 AM',
        readableDate: '9th November 2016'
      }
    ],
    membersLookup: {
      'fe27b760-a915-475c-80bb-7cdf14cc6ef3': {
        id: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
        firstName: 'Albert',
        lastName: 'Rose',
        email: 'arosec@bbb.org',
        avatar: 'http://dummyimage.com/100x100.jpg/5fa2dd/ffffff',
        ip: '20.79.231.223'
      },
      '16373df5-da0a-4074-8295-f916b94269f4': {
        id: '16373df5-da0a-4074-8295-f916b94269f4',
        firstName: 'Larry',
        lastName: 'Owens',
        email: 'lowensm@earthlink.net',
        avatar: 'http://dummyimage.com/100x100.bmp/5fa2dd/ffffff',
        ip: '168.43.167.194'
      }
    }
  }

  const tree = shallow(<MessageList {...props} />)
  expect(tree).toMatchSnapshot()
})
