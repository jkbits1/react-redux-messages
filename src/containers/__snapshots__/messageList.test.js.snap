// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with messages and members 1`] = `
<div
  style={
    Object {
      "width": 880,
    }
  }
>
  <ul>
    <Message
      key="cd445e6d-e514-424f-ba8f-16ec842002c6"
      messageDetails={
        Object {
          "id": "cd445e6d-e514-424f-ba8f-16ec842002c6",
          "message": "Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.",
          "readableDate": "9th February 2017",
          "readableTime": "4:02:00 AM",
          "timestamp": "2017-02-09T04:27:38Z",
          "userId": "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
        }
      }
      messageMember={
        Object {
          "avatar": "http://dummyimage.com/100x100.jpg/5fa2dd/ffffff",
          "email": "arosec@bbb.org",
          "firstName": "Albert",
          "id": "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
          "ip": "20.79.231.223",
          "lastName": "Rose",
        }
      }
    />
    <Message
      key="b03569ae-ccbf-4975-8040-4daba638b407"
      messageDetails={
        Object {
          "id": "b03569ae-ccbf-4975-8040-4daba638b407",
          "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
          "readableDate": "9th November 2016",
          "readableTime": "5:11:00 AM",
          "timestamp": "2016-11-09T05:04:58Z",
          "userId": "16373df5-da0a-4074-8295-f916b94269f4",
        }
      }
      messageMember={
        Object {
          "avatar": "http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
          "email": "lowensm@earthlink.net",
          "firstName": "Larry",
          "id": "16373df5-da0a-4074-8295-f916b94269f4",
          "ip": "168.43.167.194",
          "lastName": "Owens",
        }
      }
    />
  </ul>
</div>
`;

exports[`renders correctly with messages and no members 1`] = `
<div
  style={
    Object {
      "width": 880,
    }
  }
>
  <ul>
    <Message
      key="cd445e6d-e514-424f-ba8f-16ec842002c6"
      messageDetails={
        Object {
          "id": "cd445e6d-e514-424f-ba8f-16ec842002c6",
          "message": "Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla.",
          "readableDate": "9th February 2017",
          "readableTime": "4:02:00 AM",
          "timestamp": "2017-02-09T04:27:38Z",
          "userId": "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
        }
      }
      messageMember={
        Object {
          "avatar": "",
          "email": "Member info not yet loaded",
        }
      }
    />
    <Message
      key="b03569ae-ccbf-4975-8040-4daba638b407"
      messageDetails={
        Object {
          "id": "b03569ae-ccbf-4975-8040-4daba638b407",
          "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
          "readableDate": "9th November 2016",
          "readableTime": "5:11:00 AM",
          "timestamp": "2016-11-09T05:04:58Z",
          "userId": "16373df5-da0a-4074-8295-f916b94269f4",
        }
      }
      messageMember={
        Object {
          "avatar": "",
          "email": "Member info not yet loaded",
        }
      }
    />
  </ul>
</div>
`;

exports[`renders correctly with no messages or members 1`] = `""`;
